compiler/compiler.go

```diff
@@ -302,8 +302,8 @@ func compile(n semantic.Node, typeSol semantic.TypeSolution, builtIns Scope, fun
 		rt := r.Type()
 		f, err := values.LookupBinaryFunction(values.BinaryFuncSignature{
 			Operator: n.Operator,
-			Left:     lt,
-			Right:    rt,
+			Left:     lt.Nature(),
```

Comment:
Is the purpose of `Nature` to make it so that identical types compare as equal, even if they maybe represented as different objects?

Reply:
Yes, if it was using a `Type` then I would have to enumerate all array types in the map, there are an infinite number of them, so that is not possible.  Using `Nature` allows for enumerating a finite set of possible comparisons.